docs/concepts/v2/concepts/rooms/room-codes/room-code-auth.mdx

```diff
@@ -3,24 +3,184 @@ title: Room Code Authentication
 nav: 2.2
 ---
 
-[Room Codes](/concepts/v2/concepts/rooms/room-codes/room-codes) are 100ms generated short codes, representing a Room and Role combination. In addition to generating [Room Links](/concepts/v2/concepts/rooms/room-links), Room Codes can also power authentication when a peer joins a Room.
+[Room Codes](/concepts/v2/concepts/rooms/room-codes/room-codes) are 100ms generated short codes, representing a Room ID and Role combination. Room codes not only simpify authentication, they can also be used to generate [Room Links.](https://www.100ms.live/docs/concepts/v2/concepts/rooms/room-links)
 
-Primarily, joining a Room requires [Authentication Token](https://www.100ms.live/docs/concepts/v2/concepts/security-and-tokens). By using Room Codes, 100ms SDKs implicitly handle generating Authentication Token and allow a peer to join a Room without you actually generating an Auth token at your end. Note that Room Codes are not JWT tokens and thus, not entirely secure. 
+<FlexContainer>
+<Callout icon="link" title="Room Code Authentication Basics">
+- When using Room Codes, it eliminates the need to generate [auth token](/concepts/v2/concepts/Authentication/AuthToken) at your end to join a room.
+    - If you're using 100ms [Prebuilt](https://www.100ms.live/docs/concepts/v2/concepts/prebuilt) solution, Prebuilt inherently takes care of generating the auth token using room codes.
+    - When building a custom UI, you need to make an additional SDK API call `getAuthTokenByRoomCode` to generate the auth token.
+</Callout>
+</FlexContainer>
 
-Alternatively, you can always choose to authenticate a peer using Authentication Token. Read more on [how to generate Auth token](https://www.100ms.live/docs/concepts/v2/concepts/security-and-tokens#how-to-get-auth-tokens).
 
 
-## Join a Room with a Room Code
 
-**Implementation**
+## Join rooms with room codes
 
-1. Retrieve Room Code - You can get the Room Code in two ways:
+When building a custom UI and using room codes to generate room links, you can also use room codes to power authentication to join room without setting up an auth token server at your end.  
 
-    - Retrieve the Room Code from the [Room Link](/concepts/v2/concepts/rooms/room-links) that you generate.
-    - Use [Get Room Code API](/server-side/v2/api-reference/room-codes/get-room-code-api) to retrieve the Room Code
+### Implementation
 
-2. You need to then pass the Room Code in the join() method in the SDK. Here, the SDK implicitly generates Authentication Token from the Room Code.
+1. **Create Room Codes** - Room codes can be made available in two ways:
```

Comment:
this can be replaced by the <StepContainer> component - see https://100ms-docs-git-roomcodeinauth-100mslive.vercel.app/docs/concepts/v2/concepts/live-streaming as an example